src/components/ContactModal.tsx

```diff
@@ -20,27 +20,30 @@ import { CopyableAddress } from "./CopyableText";
 import { DynamicModalContext } from "./DynamicModal";
 import { FormErrorMessage } from "./FormErrorMessage";
 import colors from "../style/colors";
-import { isAddressValid } from "../types/Address";
 import { Contact } from "../types/Contact";
-import { useContactExists } from "../utils/hooks/contactsHooks";
-import { useAllAccounts, useGetOwnedAccountSafe } from "../utils/hooks/getAccountDataHooks";
+import { useValidatePkh } from "../utils/hooks/contactsHooks";
+import { useValidateName } from "../utils/hooks/labelsHooks";
 import { useAppDispatch } from "../utils/redux/hooks";
 import { contactsActions } from "../utils/redux/slices/contactsSlice";
 
+/**
+ * Modal used for both adding new contacts & editing existing contacts.
+ *
+ * Contact is checked for having unique name & pkh (among all accounts & other contacts) before being added.
+ *
+ * @param contact - optional / partial data for creating new contact, or full data for editing existing contact.
+ */
 export const UpsertContactModal: FC<{
-  title: string;
-  buttonText: string;
-  contact?: Contact; // For updating an existing contact
-}> = ({ title, buttonText, contact }) => {
+  contact?: Contact;
+}> = ({ contact }) => {
   const dispatch = useAppDispatch();
-  const getAccount = useGetOwnedAccountSafe();
   const { isOpen, onClose } = useContext(DynamicModalContext);
 
-  const onSubmitContact = (newContact: Contact) => {
-    if (getAccount(newContact.pkh)) {
-      return;
-    }
-    dispatch(contactsActions.upsert(newContact));
+  // When editing existing contact, its name & pkh are known.
+  const isEdit = contact !== undefined && contact.pkh != "" && contact.name !== "";
```

Comment:
so, if contact is not undefined then why we check that pkh and name are non-empty? they must be, don't they? also, please don't use `!=`, use `!==` instead

Reply:
The check is needed here at least for the name. 

There are 3 existing use cases for this modal. 

Creating new contact:
- undefined contact (creating from scratch)
- contact with pkh and empty name (adding used address to contacts)

Editing existing contact:
- contact with both pkh and non-empty name

Reply:
what about `const isEdit = contact?.pkh && contact?.name` then?

Reply:
Maybe I got something wrong, but it produced the wrong type
`(contact?.pkh && contact.name) ? true : false` worked though 🙂

Reply:
then `const isEdit = !!(contact?.pkh && contact.name)` please